File: src/main/java/dsalgo/interviewbits/InsertionSort.java
package dsalgo.interviewbits;

/**
 * Created by ngupta on 4/11/2019 AD.
 */
public class InsertionSort {
    public static void main(String[] args) {
        int[] arr = {5, 6, 2, 6, 9, 0, -1};
        for (int num : arr)
            System.out.print(num + ", ");
        insertionSort(arr);

        System.out.println();
        for (int num : arr)
            System.out.print(num + ", ");
    }

    private static void insertionSort(int[] arr) {
        if (arr == null || arr.length <= 0)
            return;

        for (int i = 1; i < arr.length; i++) {
            int key = arr[i];
            int j = i - 1;
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j = j - 1;
            }
            arr[j + 1] = key;
        }
    }
}

File: src/main/scala/dsalgo/interviewbit/InsertionSort.scala
package dsalgo.interviewbit

/**
 * Created by ngupta on 3/11/2019 AD.
 */
object InsertionSort extends App {
  var arr = Array(5, 6, 2, 6, 9, 0, -1)
  println(s"before sort ${arr.toSeq}")
  sort(arr)
  println(s"after sort ${arr.toSeq}")

  def sort(arr: Array[Int]) {
    for (i <- 1 until arr.length) {
      val key = arr(i)
      var j = i - 1
      while (j >= 0 && arr(j) > key) {
        arr(j + 1) = arr(j)
        j = j - 1
      }
      arr(j + 1) = key
    }
  }
}